### PR TITLE
`find_empty_slots` tag

### PR DESCRIPTION
**Pull request by BreadcrumbIsTaken**
### `<InventoryTag.find_empty_slots>`

Returns the index of each empty slot in an inventory.

Requested by [Krilliant](https://discord.com/channels/315163488085475337/1116594316903727114)

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/InventoryTag.java [hunk 1]
```diff
@@ -2356,6 +2356,28 @@ else if ((object.inventory instanceof FurnaceInventory)) {
...
+        // @returns ListTag(Number)
+        // @description
+        // Returns the index of all the empty slots in an inventory.
+        // -->
+        tagProcessor.registerTag(ListTag.class, "find_empty_slots", (attribute, object) -> {
+            if (object == null) {
+                return new ListTag();
+            }
+            ListTag indexes = new ListTag(object.getContents().length);
+            // Denizen starts at 1
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/InventoryTag.java]: This comment is stating the obvious and does not need to be here

File: plugin/src/main/java/com/denizenscript/denizen/objects/InventoryTag.java [hunk 2]
```diff
@@ -2356,6 +2356,28 @@ else if ((object.inventory instanceof FurnaceInventory)) {
...
+            int index = 1;
+            for (ItemStack itemStack : object.getContents()) {
+                if (itemStack == null) {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/InventoryTag.java]: should `|| type == air`, I don't think there's any guarantee that air items _have to become_ null.

**Comment by BreadcrumbIsTaken** [comment]
Done 👍

File: plugin/src/main/java/com/denizenscript/denizen/objects/InventoryTag.java [hunk 3]
```diff
@@ -2356,6 +2356,27 @@ else if ((object.inventory instanceof FurnaceInventory)) {
             }
             return list;
         });
+
+        // <--[tag]
+        // @attribute <InventoryTag.find_empty_slots>
+        // @returns ListTag(Number)
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/InventoryTag.java]: this should just be `ListTag`

**Pull request merged**